### ISSUE TITLE
Show message on iOS to request user to enable devicemotion manually

### ISSUE THREAD

**Issue by dmarcos** [aframevr/aframe#4149]
Since iOS 12.2 Safari `devicemotion` API comes disabled by default. The user has to enable manually on settings. We could detect iOS and if the API is disabled we can display a message above the enter VR button asking the user to enable.

**Comment by dmarcos** [comment]
It's not trivial but not super hard to implement. We could use some help if someone is interested in contributing to A-Frame

**Comment by eyexpo-dan** [comment]
How would you detect iOS safari privacy & security settings? I am listening to `devicemotion`, and it'll not trigger for either desktop, iOS 12.2+, and mobile devices without gyroscope. Is there a granular method of detection for all these possible devices?

**Comment by dmarcos** [comment]
Can just one feature detect `devicemotion`? A bit ugly but maybe:

```
 var deviceMotionEvt;
 setTimeout(function () { 
    if (!deviceMotionEvt && AFRAME.utils.isIOS()) { showMessage(); };
 }, 1000);

  window.addEventListener('devicemotion', function(event) {
   deviceMotionEvt = event;
});```

**Comment by eyexpo-dan** [comment]
Is aframe using VRDisplay in the background?

`    navigator.getVRDisplays().then((displays) => {
      // Do something with the available VR displays
    });
`

Would any of the following properties be useful for confirming if the permission is turned on or off in iOS?

`
isIOS: true
gyroscope { x: 0, y: 0, z: 0 }
sensor: null
errors: ... Can't find RelativeOrientationSensor
`

**Comment by dmarcos** [comment]
Whatever works. In my snippet above I assumed the only way to detect devicemotion is listening to events that would always fire at page load if the API is enabled. A synchronous way would be better if there’s any.

**Comment by EslamNemr** [comment]
using callback
https://aframe-ios-motion-orientation-access.glitch.me/
tried on i phone xs-max
can we use promise ?

**Comment by dmarcos** [comment]
not sure If I understood. devicemotion is not promise API

**Comment by EslamNemr** [comment]
I mean Async/Await to avoid timeout.

**Comment by micsun-al** [comment]
iPad Air (MD785LL/A) running iOS 11.4 (15F79) laying flat on a table does not send any `devicemotion` events until the device actually moves, so this setTimeout devicemotion check fails ... wonder what other things we can check for hrmmm (time for more googling).

**Comment by dmarcos** [comment]
Maybe is not that big of a deal. Is that common to browse on a tablet flat on a table? One could hide the message once as soon as the event fires that would happen as soon as the tablet is moved.

**Comment by dmarcos** [comment]
Example on how 8th wall deals with the issue: https://github.com/aframevr/aframe/issues/3976#issuecomment-493035158

**Comment by dmarcos** [comment]
I think a message is not necessary anymore. Disabling devicemotion and relying on user settings was an interim solution. iOS 13 beta has a `DeviceOrientationEvent.requestPermission` that we should use. It will prompt the user for permission.

**Comment by jamesdigid** [comment]
> I think a message is not necessary anymore. Disabling devicemotion and relying on user settings was an interim solution. iOS 13 beta has a `DeviceOrientationEvent.requestPermission` that we should use. It will prompt the user for permission.

That seems ideal, it also appears that in Chrome on my iPhone X (iOS 12.3.1) with tracking turned off it is still completely functional in Chrome. Talk about security... If there are any other small issues you could recommend that could get me familiar with the code would be much appreciated. I'm still learning the syntactic sugar coming mostly from Python, and Java codebases.

**Comment by brianpeiris** [comment]
Closed by #4303?

**Comment by dmarcos** [comment]
Correct. Thank you 🙏
